gems/pending/appliance_console/service_group.rb

```diff
@@ -1,64 +0,0 @@
-require "appliance_console/internal_database_configuration"
-require "util/postgres_admin"
-
-module ApplianceConsole
-  class ServiceGroup
-    SERVICES  = %w(evminit memcached miqtop evmserverd).freeze
-
-    def initialize(hash = {})
-      @postgresql = hash[:internal_postgresql]
-    end
-
-    def postgresql?
-      !!@postgresql
-    end
-
-    def restart_services
-      enablement
-      restart
-    end
-
-    def enablement
-      disable
-      enable
-    end
-
-    def restart
-      stop
-      start
-    end
-
-    def enable
-      enable_miqtop
-      SERVICES.each { |s| run_service(s, "enable") }
-      run_service(PostgresAdmin.service_name, "enable") if postgresql?
-    end
-
-    def disable
-      run_service(PostgresAdmin.service_name, "disable") unless postgresql?
-    end
-
-    def start
-      SERVICES.each { |s| run_detached_service(s, "start") }
-    end
-
-    def stop
-      run_service(PostgresAdmin.service_name, "stop") unless postgresql?
-    end
-
-    private
-
-    def enable_miqtop
-      LinuxAdmin::Service.new("miqtop").enable
-    end
-
-    def run_service(service, action)
-      LinuxAdmin::Service.new(service).send(action)
-    end
-
-    # TODO: Fix LinuxAdmin::Service to detach.
-    def run_detached_service(service, action)
-      Process.detach(Kernel.spawn("/sbin/service #{service} #{action}", [:out, :err] => ["/dev/null", "w"]))
```

Comment:
It turns out doing this was solving a deadlock in our systemd service file used to create the database on first boot.

The [service](https://github.com/ManageIQ/manageiq-appliance-build/blob/master/kickstarts/partials/post/db_init.ks.erb#L17) we create that initializes the database does so by running `appliance_console_cli` which in turn will attempt to start the `evmserverd` unit. Because that service (appliance-initialize) is a "oneshot" unit it will wait until the command finishes before considering the service "started". That unit also specifies `Before=evmserverd.service` which will [delay the evmserverd unit from starting until this on (appliance-initialize) has started.](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#Before=)

Because of this, the appliance-initialize service will hang and evmserverd will not be started.